code/game/machinery/vending/vendor_types/requisitions.dm

```diff
@@ -133,9 +133,30 @@
 		)
 
 /obj/structure/machinery/cm_vending/sorted/cargo_guns/stock(obj/item/item_to_stock, mob/user)
+	if(istype(item_to_stock, /obj/item/storage) && !istype(item_to_stock, /obj/item/storage/box/m94) && !istype(item_to_stock, /obj/item/storage/large_holster/machete))
+		to_chat(user, SPAN_WARNING("Can't restock \the [item_to_stock]."))
+		return
+
+	//this below is in case we have subtype of an object, that SHOULD be treated as parent object (like /empty ammo box)
+	var/corrected_path = return_corresponding_type(item_to_stock.type)
+
 	var/list/R
 	for(R in (listed_products))
-		if(item_to_stock.type == R[3] && !istype(item_to_stock,/obj/item/storage))
+		if(item_to_stock.type == R[3] || corrected_path && corrected_path == R[3])
+			//magazines handling (mainly for subtypes)
+			if(istype(item_to_stock, /obj/item/ammo_magazine))
+				//flamer fuel tanks handling
+				if(istype(item_to_stock, /obj/item/ammo_magazine/flamer_tank))
+					var/obj/item/ammo_magazine/flamer_tank/FT = item_to_stock
+					if(FT.flamer_chem != initial(FT.flamer_chem))
+						to_chat(user, SPAN_WARNING("\The [FT] contains not standard fuel."))
```

Comment:
```suggestion
						to_chat(user, SPAN_WARNING("\The [FT] contains non-standard fuel."))
```
there MUST be a better way to handle this than repeating the same code three times

possibly a can_restock proc on the item that returns true or false?